src/plugins/dashboard/public/dashboard_container/embeddable/dashboard_container.tsx

```diff
@@ -331,10 +332,21 @@ export class DashboardContainer extends Container<InheritedChildInput, Dashboard
     const {
       explicitInput: { timeRange, refreshInterval },
       componentState: {
-        lastSavedInput: { timeRestore: lastSavedTimeRestore },
+        lastSavedInput: {
+          controlGroupInput: lastSavedControlGroupInput,
+          timeRestore: lastSavedTimeRestore,
+        },
       },
     } = this.getState();
 
+    if (
+      this.controlGroup &&
+      lastSavedControlGroupInput &&
+      !persistableControlGroupInputIsEqual(this.controlGroup.getInput(), lastSavedControlGroupInput)
+    ) {
+      this.controlGroup.updateInput(lastSavedControlGroupInput);
+    }
```

Comment:
It's more reliable + predictable to simply reset the control group directly here rather than relying on a subscription to listen for changes to the Dashboard's control group input.